helper/metricsutil/wrapped_metrics_test.go

```diff
@@ -0,0 +1,114 @@
+package metricsutil
+
+import (
+	"testing"
+	"time"
+
+	"github.com/armon/go-metrics"
+)
+
+func isLabelPresent(toFind Label, ls []Label) bool {
+	for _, l := range ls {
+		if l == toFind {
+			return true
+		}
+	}
+	return false
+}
+
+func TestClusterLabelPresent(t *testing.T) {
+	// With only one test method, we can check that sending a metric
+	// before SetGlobalSink doesn't crash anything.
+	// If we had multiple test methods, I don't know how we would check that
+	// since we can't guarantee order?
+
+	SetGaugeWithLabels([]string{"test", "gauge"},
+		1.0,
+		[]Label{
+			{"label", "value"},
+		},
+	)
+
+	testClusterName := "test-cluster"
+
+	// Use a ridiculously long time to minimize the chance
+	// that we have to deal with more than one interval.
+	// InMemSink rounds down to an interval boundary rather than
+	// starting one at the time of initialization.
+	inmemSink := metrics.NewInmemSink(
+		1000000*time.Hour,
+		2000000*time.Hour)
+	clusterSink := &ClusterMetricSink{
+		ClusterName: testClusterName,
+		Sink:        inmemSink,
+	}
+
+	SetGlobalSink(clusterSink)
+
+	key1 := []string{"aaa", "bbb"}
+	key2 := []string{"ccc", "ddd"}
+	key3 := []string{"eee", "fff"}
+	labels1 := []Label{{"dim1", "val1"}}
+	labels2 := []Label{{"dim2", "val2"}}
+	labels3 := []Label{{"dim3", "val3"}}
+	clusterLabel := Label{"cluster", testClusterName}
+	expectedKey1 := "aaa.bbb;dim1=val1;cluster=" + testClusterName
+	expectedKey2 := "ccc.ddd;dim2=val2;cluster=" + testClusterName
+	expectedKey3 := "eee.fff;dim3=val3;cluster=" + testClusterName
+
+	SetGaugeWithLabels(key1, 1.0, labels1)
+	IncrCounterWithLabels(key2, 2.0, labels2)
+	AddSampleWithLabels(key3, 3.0, labels3)
+
+	intervals := inmemSink.Data()
+	if len(intervals) > 1 {
+		t.Log("Skipping test, detected interval crossing.")
```

Comment:
Probably worth a little comment explaining this bit.  My understanding is it's a very rare event that you haven't bothered to handle, which is fine with me.  Also consider `t.SkipNow()` instead of `return`, that way it should show up in CI more obviously.

Reply:
Added a comment. If we're close to an interval boundary the three metrics could end up split across two buckets; setting the interval to 100,000 hours makes this extremely unlikely.

Thanks for pointing me at SkipNow(), changed.